pstatus-report-sink-ktor/src/main/kotlin/gov/cdc/ocio/processingstatusapi/plugins/AWSSQS.kt

```diff
@@ -74,33 +70,39 @@ val AWSSQSPlugin = createApplicationPlugin(
         SchemaValidation.logger.info("Consuming messages from AWS SQS")
         CoroutineScope(Dispatchers.IO).launch {
             while (true) {
+                var response: ReceiveMessageResponse? = null
                 try {
                     val receiveMessageRequest = ReceiveMessageRequest {
                         this.queueUrl = queueUrl
                         maxNumberOfMessages = 5
                     }
-                    val response = sqsClient.receiveMessage(receiveMessageRequest)
+                    response = sqsClient.receiveMessage(receiveMessageRequest)
                     response.messages?.forEach { message ->
                         SchemaValidation.logger.info("Received message from AWS SQS: ${message.body}")
-
                         message.body?.let { AWSSQSProcessor().validateMessage(it) }
-                        val deleteMessageRequest = DeleteMessageRequest {
-                            this.queueUrl = queueUrl
-                            this.receiptHandle = message.receiptHandle
-                        }
-                        sqsClient.deleteMessage(deleteMessageRequest)
-                        SchemaValidation.logger.info("Successfully Deleted processed message from AWS SQS")
-
                     }
                 } catch (e: AwsServiceException) {
                     SchemaValidation.logger.error("Something went wrong while processing the request ${e.message}")
+                    throw e
                 } catch (e: Exception) {
                     SchemaValidation.logger.error("AWS service exception occurred: ${e.message}")
+                    throw e
+                }finally {
+                    response?.messages?.forEach { message ->
+                        try {
+                            val deleteMessageRequest = DeleteMessageRequest {
+                                this.queueUrl = queueUrl
+                                this.receiptHandle = message.receiptHandle
+                            }
+                            sqsClient.deleteMessage(deleteMessageRequest)
+                            SchemaValidation.logger.info("Successfully Deleted processed message from AWS SQS")
```

Comment:
Are these for "processed" messages? The verbiage doesn't seem to be right

Reply:
Message arrived at the queue, processed from the queue and then deleted. do you propose processed report instead?